src/Pimple/Container.php

```diff
@@ -113,7 +114,14 @@ public function offsetGet($id)
         $val = $this->values[$id] = $raw($this);
         $this->raw[$id] = $raw;
 
-        $this->frozen[$id] = true;
+        if (!isset($this->frozen[$id])) {
```

Comment:
this check is not needed. This code cannot be reached again if the object was already created earlier. It would have been returned at line 106 already.